newIDE/app/src/ObjectsList/index.js

```diff
@@ -638,27 +638,62 @@ const ObjectsList = React.forwardRef<Props, ObjectsListInterface>(
 
     const onObjectsAddedFromAssets = React.useCallback(
       (objects: Array<gdObject>) => {
+        if (objects.length === 0) return;
+
         objects.forEach(object => {
           onObjectCreated(object);
         });
-        if (treeViewRef.current)
-          treeViewRef.current.openItems([sceneObjectsRootFolderId]);
 
         const lastObject = objects[objects.length - 1];
-        // A new object is always added to the scene (layout) by default.
-        const object = objectsContainer
-          .getRootFolder()
-          .getObjectChild(lastObject.getName());
-
+        let objectToScrollTo;
+        if (
+          newObjectDialogOpen &&
+          newObjectDialogOpen.from &&
+          // If a scene objectFolderOrObject is selected, move added assets next to or inside it.
+          !newObjectDialogOpen.from.global
+        ) {
+          const selectedItem = newObjectDialogOpen.from.objectFolderOrObject;
+          if (treeViewRef.current) {
+            treeViewRef.current.openItems(
+              getFoldersAscendanceWithoutRootFolder(selectedItem).map(folder =>
+                getObjectFolderTreeViewItemId(folder)
+              )
+            );
+          }
+          const {
+            folder: parentFolder,
+            position,
+          } = getInsertionParentAndPositionFromSelection(selectedItem);
+          const rootFolder = objectsContainer.getRootFolder();
+          objects.forEach((object, index) => {
+            const objectFolderOrObject = rootFolder.getObjectChild(
+              object.getName()
+            );
+            rootFolder.moveObjectFolderOrObjectToAnotherFolder(
```

Comment:
instead of moving them after they're created, should we instead modify the  `targetObjectsContainer` to be the selected folder in the `AssetPackInstallDialog` instead of the root of the scene?